src/pages/Home/Home.style.ts

```diff
@@ -0,0 +1,13 @@
+import styled from "styled-components";
+
+type Props = {
+  menuOpen: boolean;
+};
+
+export const HomeStyle = styled.div<Props>`
```

Comment:
Reason why this method does not work:
- When the modal is open, you set the height to a fixed value that is shorter than the automatic height of the element and set overflow-y: false, this means that elements that are out of the 100vh height will be hidden. Therefore, when you open the modal, the other components will be hidden from the user and the viewport is scrolled to the top
- You cannot have the modal in the same component as the Home

Solution
- Create a react portal to another sibling div to #root and set this new div's z-index to be higher than root. You can use react-transition-group to do the transitions for that, but for now you can leave it first since transitions aren't so important for now
- Go and refer to the react course and read docs/watch videos to learn how to create a portal
- Regarding the scrolling part, when the modal is open, you can add this class to the body. Then you remove it when the modal is removed

.stop-scrolling {
	height: 100%;
	overflow: hidden;
}

Conclusion: The overlay and the menu should be in a div which is inserted to the DOM via react-portal. This div should have higher z-index than the #root

Reply:
Fixed.